helpers/metadata.go

```diff
@@ -33,3 +38,37 @@ func HasLabelWithValue(object client.Object, label, value string) bool {
 
 	return false
 }
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the specified prefix.
+// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
+func CopyLabelsByPrefix(src *metav1.ObjectMeta, dest *metav1.ObjectMeta, prefix string, replacementPrefix string) {
+	if dest.Labels == nil {
+		dest.Labels = make(map[string]string)
+	}
+	for key, value := range src.GetLabels() {
+		if strings.HasPrefix(key, prefix) {
+			newKey := key
+			if prefix != replacementPrefix {
+				newKey = strings.Replace(key, prefix, replacementPrefix, 1)
+			}
+			dest.Labels[newKey] = value
+		}
+	}
```

Comment:
Moving this code into a helper function was a good move, it makes the pipeline_adapter.go look clean. :smiley: 
Now consider using another function to do the things common to both **Labels** and **Annotations**.

Something like this:
```
// copyWithNewPrefix copies key/value pairs in the source map adding them into the destination map.
func copyWithNewPrefix(src, dest map[string]string, prefix, replacementPrefix string) {
	for key, value := range src {
		if strings.HasPrefix(key, prefix) {
			newKey := key
			if prefix != replacementPrefix {
				newKey = strings.Replace(key, prefix, replacementPrefix, 1)
			}
			dest[newKey] = value
		}
	}
}
```

Then you can make both these functions a wrapper for the underlying function. Like so...
```
// CopyAnnotationsByPrefix copies all annotations from a source object to a destination object where the key matches the specified prefix.
// If replacementPrefix is different from prefix, the prefix will be replaced while performing the copy.
func CopyAnnotationsByPrefix(src, dest *metav1.ObjectMeta, prefix, replacementPrefix string) {
	if dest.GetAnnotations() == nil {
		dest.SetAnnotations(map[string]string{})
	}
	copyWithNewPrefix(src.GetAnnotations(), dest.GetAnnotations(), prefix, replacementPrefix)
}
```
That way you only have to write the boilerplate code once, test in a single place at the helper function, and the two wrapper function only need to test they created maps if they are nil, and only operate on the given **Label** or **Annotation**.

Reply:
Thanks for the suggestion! Applied.